homeassistant/components/unifiprotect/binary_sensor.py

```diff
@@ -682,21 +671,19 @@ def __init__(
         super().__init__(data, device, description)
 
     @callback
-    def _async_update_device_from_protect(self, device: ProtectModelWithId) -> None:
+    def _async_update_device_from_protect(self, device: ProtectDeviceType) -> None:
         super()._async_update_device_from_protect(device)
         slot = self._disk.slot
-        self._attr_available = False
-        available = self.data.last_update_success
-
+        ustorage = self.device.system_info.ustorage
         # should not be possible since it would require user to
         # _downgrade_ to make ustorage disppear
-        assert self.device.system_info.ustorage is not None
-        for disk in self.device.system_info.ustorage.disks:
-            if disk.slot == slot:
-                self._disk = disk
-                self._attr_available = available
-                break
-
+        assert ustorage is not None
+        disk = next((d for d in ustorage.disks if d.slot == slot), None)
+        if disk is None:
+            # disk was removed
+            self._attr_available = False
+            return
```

Comment:
I'll take this out and clean it up in a different PR as this is technically a bugfix